In [1]:
import pandas as pd
from utils import get_Xy, preprocess, model_generator, evaluate

In [2]:
%load_ext autoreload
%autoreload 2

In [3]:
df = pd.read_csv('../data/predictive_maintenance.csv')

In [4]:
X_train, X_test, y_train, y_test = get_Xy(df, 'Failure Type', binary=['No Failure'], drop_cols=['UDI', 'Product ID'])

In [5]:
preprocessor = preprocess(X_train, y_train)

In [6]:
pipeline = model_generator(X_train, y_train, preprocessor, models=None)

Fitting 5 folds for each of 80 candidates, totalling 400 fits
[CV 1/5] END leaf_size=10, n_neighbors=1, weights=uniform;, score=0.994 total time=   0.4s
[CV 2/5] END leaf_size=10, n_neighbors=1, weights=uniform;, score=0.995 total time=   0.2s
[CV 3/5] END leaf_size=10, n_neighbors=1, weights=uniform;, score=0.996 total time=   0.2s
[CV 4/5] END leaf_size=10, n_neighbors=1, weights=uniform;, score=0.995 total time=   0.2s
[CV 5/5] END leaf_size=10, n_neighbors=1, weights=uniform;, score=0.995 total time=   0.2s
[CV 1/5] END leaf_size=10, n_neighbors=1, weights=distance;, score=0.994 total time=   0.2s
[CV 2/5] END leaf_size=10, n_neighbors=1, weights=distance;, score=0.995 total time=   0.2s
[CV 3/5] END leaf_size=10, n_neighbors=1, weights=distance;, score=0.996 total time=   0.2s
[CV 4/5] END leaf_size=10, n_neighbors=1, weights=distance;, score=0.995 total time=   0.2s
[CV 5/5] END leaf_size=10, n_neighbors=1, weights=distance;, score=0.995 total time=   0.2s
[CV 1/5] END leaf_size=

In [7]:
evals = evaluate(pipeline, X_test, y_test)

In [17]:
for e in evals:
    print(e['model'], '-' * (54 - len(e['model'])))
    print(e['report'])

Logistic Regression -----------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.96      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.98      0.98      2000
weighted avg       1.00      1.00      1.00      2000

KNN ---------------------------------------------------
              precision    recall  f1-score   support

    negative       0.96      0.89      0.93        76
    positive       1.00      1.00      1.00      1924

    accuracy                           0.99      2000
   macro avg       0.98      0.95      0.96      2000
weighted avg       0.99      0.99      0.99      2000

Decision Tree -----------------------------------------
              precision    recall  f1-score   support

    negative       0.93      0.93      0.93        76
    positive       1.00      1.00      1.00      1924

    accuracy  

Logistic Regression -----------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.96      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.98      0.98      2000
weighted avg       1.00      1.00      1.00      2000

KNN ---------------------------------------------------
              precision    recall  f1-score   support

    negative       0.96      0.89      0.93        76
    positive       1.00      1.00      1.00      1924

    accuracy                           0.99      2000
   macro avg       0.98      0.95      0.96      2000
weighted avg       0.99      0.99      0.99      2000

Decision Tree -----------------------------------------
              precision    recall  f1-score   support

    negative       0.93      0.93      0.93        76
    positive       1.00      1.00      1.00      1924

    accuracy                           0.99      2000
   macro avg       0.97      0.97      0.97      2000
weighted avg       0.99      0.99      0.99      2000

Random Forest -----------------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.96      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.98      0.98      2000
weighted avg       1.00      1.00      1.00      2000

Extremely Random Trees --------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.96      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.98      0.98      2000
weighted avg       1.00      1.00      1.00      2000

Gradient Boosting -------------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.96      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.98      0.98      2000
weighted avg       1.00      1.00      1.00      2000

AdaBoost ----------------------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.95      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.97      0.97      2000
weighted avg       1.00      1.00      1.00      2000

SVM ---------------------------------------------------
              precision    recall  f1-score   support

    negative       0.95      0.96      0.95        76
    positive       1.00      1.00      1.00      1924

    accuracy                           1.00      2000
   macro avg       0.97      0.98      0.98      2000
weighted avg       1.00      1.00      1.00      2000

